dockstore-webservice/src/main/java/io/dockstore/webservice/CloudWatchMetricsReporter.java

```diff
@@ -0,0 +1,87 @@
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.codahale.metrics.Timer;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration.ExternalConfig;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import software.amazon.awssdk.services.cloudwatch.model.MetricDatum;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataRequest;
+import software.amazon.awssdk.services.cloudwatch.model.StandardUnit;
+
+public class CloudWatchMetricsReporter extends ScheduledReporter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CloudWatchMetricsReporter.class);
+    private CloudWatchClient cw = null;
+    private String namespace = null;
+
+    protected CloudWatchMetricsReporter(MetricRegistry registry, String name, MetricFilter filter, TimeUnit rateUnit,
+        TimeUnit durationUnit, ExternalConfig config) {
+        super(registry, name, filter, rateUnit, durationUnit);
+        initialize(config);
+    }
+
+    private void initialize(ExternalConfig config) {
+        try {
+            this.cw = CloudWatchClient.builder()
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build();
+            this.namespace = config.getHostname() + "_LogMetrics";
+        } catch (SdkClientException e) {
+            LOG.error("Unable to create CloudWatchClient", e);
+        }
+    }
+
+    @Override
+    public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters, SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters,
+        SortedMap<String, Timer> timers) {
+
+        // start with Guages but we should add the rest later
+        // inspired from https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/java_cloudwatch_code_examples.html
+        List<MetricDatum> metricDataList = new ArrayList<>();
+        gauges.forEach((key, value) -> {
+            String time = ZonedDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+            Instant instant = Instant.parse(time);
+            MetricDatum metricDatum = getMetricDatum(key, Double.parseDouble(value.getValue().toString()), instant);
+            metricDataList.add(metricDatum);
+        });
+
+        PutMetricDataRequest request = PutMetricDataRequest.builder()
+            .namespace(namespace)
+            .metricData(metricDataList)
+            .build();
+
+        try {
+            cw.putMetricData(request);
```

Comment:
I think the rule of thumb I learned was more not to throw exceptions for regular control flow. But since this is an exceptional situation ...  that said, I should probably not catch the basic `Exception` class, so will eliminate both.